File: OCExample/OCExample/LTFollowMoveView/LTFollowMoveDemo.m
//
//  LTFollowMoveDemo.m
//  OCExample
//
//  Created by gaoliutong on 2022/3/5.
//  Copyright © 2022 LT. All rights reserved.
//
//  如有疑问，请搜索并关注微信公众号"技术大咖社"并留言即可
//
//  ScrollView嵌套ScrolloView解决方案（初级、进阶)， 支持OC/Swift
//
//  github地址: https://github.com/gltwy/LTScrollView
//
//  clone地址:  https://github.com/gltwy/LTScrollView.git
//

#import "LTFollowMoveDemo.h"
#import "LTFollowMoveTestView.h"
#import "LTScrollView-Swift.h"

@interface LTFollowMoveDemo ()
@property(strong, nonatomic) UIButton *jumpButton;
@property(strong, nonatomic) LTFollowMoveTestView *testView;
@end

@implementation LTFollowMoveDemo

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor grayColor];
    [self.view addSubview:self.jumpButton];
}

- (void)jumpButtonClick {
    LTFollowMoveView *moveView = [[LTFollowMoveView alloc] initWithFrame:UIScreen.mainScreen.bounds contentView:self.testView];
    [moveView setShowFinishedHandle:^{
        NSLog(@"setShowFinishedHandle");
    }];
    [moveView showInView:self.view];
}

- (LTFollowMoveTestView *)testView {
    if (!_testView) {
        _testView = [[LTFollowMoveTestView alloc] initWithFrame:CGRectMake(0, 0, self.view.frame.size.width, kReturnHeight)];
    }
    return _testView;
}

- (UIButton *)jumpButton {
    if (!_jumpButton) {
        _jumpButton = [UIButton buttonWithType:UIButtonTypeCustom];
        _jumpButton.frame = CGRectMake(0, kReturnHeight, self.view.frame.size.width, 50);
        [_jumpButton setTitle:@"点击弹出" forState:UIControlStateNormal];
        _jumpButton.titleLabel.font = [UIFont systemFontOfSize:14];
        [_jumpButton addTarget:self action:@selector(jumpButtonClick) forControlEvents:UIControlEventTouchUpInside];
        _jumpButton.backgroundColor = [UIColor redColor];
    }
    return _jumpButton;
}

@end

File: Example/LTScrollView/Demo/LTFollowMoveView/LTFollowMoveDemo.swift
//
//  LTFollowMoveDemo.swift
//  LTScrollView_Example
//
//  Created by gaoliutong on 2022/3/5.
//  Copyright © 2022 CocoaPods. All rights reserved.
//

import UIKit

class LTFollowMoveDemo: UIViewController, LTFollowMoveViewDelegate {

    private lazy var cancelButton: UIButton = {
        let cancelButton = UIButton(type: .custom)
        cancelButton.frame = CGRect(x: 0, y: returnHeight, width: view.glt_width, height: 50)
        cancelButton.setTitle("点击弹出", for: .normal)
        cancelButton.backgroundColor = .red
        cancelButton.addTarget(self, action: #selector(jump), for: .touchUpInside)
        return cancelButton
    }()
    
    private lazy var moveView: LTFollowMoveTestView = {
        let moveView = LTFollowMoveTestView(frame: CGRect(x: 0, y: 0, width: self.view.glt_width, height: returnHeight))
        return moveView
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .gray
        view.addSubview(cancelButton)
    }
    
    @objc func jump() {
        let testView = LTFollowMoveView(frame: UIScreen.main.bounds, contentView: self.moveView)
        testView.delegate = self
        /**
        testView.stretchRate = 0.5
        testView.isTapHide = true
        testView.isQuickPanHide = false
        testView.isCanPanBlank = true
        testView.willShowHandle = {
            print("即将展示 - 1")
        }
        testView.showFinishedHandle = {
            print("展示完成 - 1")
        }
        testView.dismissFinishedHandle = {
            print("消失完成 - 1")
        }
         */
        testView.show(inView: self.view)
    }
    
    func showFinished(followView: LTFollowMoveView) {
        print("展示完成 - 2")
    }
    
    func dismissFinished(followView: LTFollowMoveView) {
        print("消失完成 - 2")
    }
    
    func willShow(followView: LTFollowMoveView) {
        print("即将展示 - 2")
    }
    
}
